akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -416,4 +406,10 @@ class QueueSourceSpec extends StreamSpec {
 
   }
 
+  "QueueOfferResult" should {
+    "be enqueued for Enqueued" in {
+      enqueued.isSuccess shouldBe (true)
+    }
```

Comment:
Add the cases for other types I mentioned in the previous comment and you will see that you implementation is still not correct.